File: Sample/BbQ.ChatWidgets.Sample.React/Models/WeatherWidget.cs
using BbQ.ChatWidgets.Models;

namespace BbQ.ChatWidgets.Sample.WebApp.Models;

/// <summary>
/// Server-side weather widget used for the SSE demo.
/// Displays current weather conditions pushed via SSE with real-time updates.
/// </summary>
public sealed record WeatherWidget(
    string Label,
    string Action,
    string? City = null,
    string? StreamId = null
) : ChatWidget(Label, Action)
{
    public override string Purpose => """
           **Weather Widget** - Displays weather data pushed via SSE
           Emitted through the schema-constrained emit_widgets tool as complete weather state.
        """;
}

File: Sample/BbQ.ChatWidgets.Sample.React/ClientApp/src/widgets/WeatherWidget.ts
import { ChatWidget } from '@bbq/chatwidgets';

/**
 * Weather widget model for client-side rendering.
 * Extends ChatWidget with weather-specific properties and SSE support.
 */
export class WeatherWidget extends ChatWidget {
  public readonly city?: string;
  public readonly streamId?: string;

  constructor(label: string, action: string, city?: string, streamId?: string) {
    super('weather', label, action);
    this.city = city;
    this.streamId = streamId;
  }

  toObject(): any {
    return {
      type: 'weather',
      label: this.label,
      action: this.action,
      city: this.city,
      streamId: this.streamId,
    };
  }
}
